### ISSUE TITLE
this.elemMods inconsistency between versions

### ISSUE THREAD

**Issue by tadatuta** [bem/bem-xjst#75]
```js
match(function() { return this.elemMods.interactive === 'yes'; })
```
with `bem-xjst 1.x` is ok even for elems without any modifiers and fails on `bem-xjst 2.x` with `TypeError: Cannot read property 'interactive' of undefined`

So it looks like `elemMods` is not explicitly created.
Not sure if it should be fixed.

// cc @veged @indutny

**Comment by indutny** [comment]
This was done for performance reasons, creating empty `elemMods` per each block is potentially expensive. cc @veged

**Comment by veged** [comment]
we doesn't need to do it for each __block__, just for each __element__

also we can add `elemMods = {}` to `BEMContext.prototype` and override it when entering to element with own `elemMods`

**Comment by indutny** [comment]
@veged I'm -1 for doing for each element, but definitely will do it at the BEMContext init.

**Comment by indutny** [comment]
Fixed in 56d9d5f. Thanks!
